packages/devtools-map-bindings/src/remapScopes.js

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import type {
+  Scope,
+  SourceScope,
+  MappedScopeBindings,
+  SyntheticScope
+} from "debugger-html";
+
+export type RemappedScope = {
+  scopes: SyntheticScope[],
+  start: number,
+  end: number
+};
+
+// Performs mapping of the original parsed scopes to the locals mappings
+// based on the generated source parse and source map data.
+function remapScopes(
+  scopes: ?(SourceScope[]),
+  generatedScopes: MappedScopeBindings[]
+): ?(RemappedScope[]) {
+  if (!scopes || scopes.length === 0) {
+    return null;
+  }
+  // Chunk original source scopes on function/closure boundary
+  const { result: scopeChunks } = scopes.reduce(
+    ({ isLast, result }, scope) => {
+      if (isLast) {
+        result.push([]);
+      }
```

Comment:
chunk is a pretty strange term...

I think i would prefer `functionScopes`,  perhaps we move this into a small helper function:

```js
const functionScopes = rollupFunctionScopes(scopes)
```